lib/identity/cookie_coder.rb

```diff
@@ -0,0 +1,58 @@
+require "base64"
+
+module Identity
+  # CookieCoder encodes and decodes payload before setting it to cookie
+  # Different from FernetCookieCoder, it doesn't use Ruby Marshal.
+  # The encrypted cookie is meant to consumed by other programming languages other than Ruby.
+  class CookieCoder
```

Comment:
The name of the class is more specific than it needs to be. There is nothing specific to cookies in the implementation. It's `HashCoder` by implementation and a `CookieCoder` only due to its usage.  The class name doesn't have to change but, if being able to encode a Hash is a feature we may need elsewhere, it may be better to name the class for what it is.